File: cs/Minsk/CodeAnalysis/Binding/BoundAssignmentExpression.cs
namespace Minsk.CodeAnalysis.Binding;

internal sealed class BoundAssignmentExpression : BoundExpression
{
    public BoundAssignmentExpression(VariableSymbol variable, BoundExpression expression)
    {
        Variable = variable;
        Expression = expression;
    }

    public VariableSymbol Variable { get; }
    public BoundExpression Expression { get; }
    public override BoundNodeKind Kind => BoundNodeKind.AssignmentExpression;
    public override Type Type => Expression.Type;
}

File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/binding/BoundAssignmentExpression.kt
package dev.phytolizer.minsk.analysis.binding

import dev.phytolizer.minsk.analysis.VariableSymbol
import kotlin.reflect.KClass

internal class BoundAssignmentExpression(val variable: VariableSymbol, val expression: BoundExpression) :
    BoundExpression() {
    override val kind: BoundNodeKind
        get() = BoundNodeKind.AssignmentExpression
    override val type: KClass<out Any>
        get() = variable.type
}
